### ISSUE TITLE
Fails to load on MC 1.20.2

### ISSUE THREAD

**Issue by SlimeDog** [RockinChaos/ItemJoin#552]
### Server Version

Paper 1.20.2-201 or Spigot 1.20.2-3888

### Resource Version

ItemJoin 6.0.3-SNAPSHOT-b901

### Dump Information

https://pastebin.com/66TEvz3X

### Describe The Bug

```
[10:39:54] [Server thread/INFO]: [ItemJoin] Loading server plugin ItemJoin v6.0.3-SNAPSHOT-b901
...
[10:40:07] [Server thread/INFO]: [ItemJoin] Enabling ItemJoin v6.0.3-SNAPSHOT-b901
...
[10:40:43] [Craft Scheduler Thread - 6 - ItemJoin/INFO]: [ItemJoin] Running a developer version ... skipping NMS check.
[10:40:43] [Craft Scheduler Thread - 6 - ItemJoin/INFO]: [ItemJoin] Hooked into { Multiverse-Core, Multiverse-Inventories, WorldGuard, PlaceholderAPI, ProtocolLib, Vault }
[10:40:43] [Craft Scheduler Thread - 6 - ItemJoin/INFO]: [ItemJoin] 13 Custom item(s) loaded!
...
[10:40:44] [Server thread/WARN]: [ItemJoin] Task #24 for ItemJoin v6.0.3-SNAPSHOT-b901 generated an exception
java.lang.NullPointerException: Profile name must not be null
        at java.util.Objects.requireNonNull(Objects.java:259) ~[?:?]
        at com.mojang.authlib.GameProfile.<init>(GameProfile.java:31) ~[authlib-5.0.47.jar:?]
        at me.RockinChaos.itemjoin.item.ItemDesigner.setSkullTexture(ItemDesigner.java:932) ~[ItemJoin-6.0.3-b901.jar:?]
        at me.RockinChaos.itemjoin.item.ItemDesigner.registerItems(ItemDesigner.java:113) ~[ItemJoin-6.0.3-b901.jar:?]
        at me.RockinChaos.itemjoin.item.ItemDesigner.lambda$new$1(ItemDesigner.java:77) ~[ItemJoin-6.0.3-b901.jar:?]
        at org.bukkit.craftbukkit.v1_20_R2.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.20.2.jar:git-Paper-201]
        at org.bukkit.craftbukkit.v1_20_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:480) ~[paper-1.20.2.jar:git-Paper-201]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1097) ~[paper-1.20.2.jar:git-Paper-201]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:315) ~[paper-1.20.2.jar:git-Paper-201]
        at java.lang.Thread.run(Thread.java:1583) ~[?:?]
```

### Expected Behavior

Clean start-up

### Steps To Reproduce

- Install Paper 1.20.2-201
- Install ItemJoin 6.0.3-SNAPSHOT-b901
- Start server
- Read log

### Checklist

- [X] I am running latest version of this resource.
- [X] I have read the wiki to make sure it's not an issue with configuration.
- [ ] I ticked all of these boxes without actually reading them.
- [X] I have checked that my bug report is unique and a previous report does not exist.

**Comment by SlimeDog** [comment]
Paper 1.20.2-201
ItemJoin 6.0.3-EXPERIMENTAL-b905

On server start-up
```
[15:58:47] [Server thread/WARN]: [ItemJoin] Task #28 for ItemJoin v6.0.3-EXPERIMENTAL-b905 generated an exception
java.lang.NullPointerException: Profile name must not be null
        at java.util.Objects.requireNonNull(Objects.java:259) ~[?:?]
        at com.mojang.authlib.GameProfile.<init>(GameProfile.java:31) ~[authlib-5.0.47.jar:?]
        at me.RockinChaos.itemjoin.item.ItemDesigner.setSkullTexture(ItemDesigner.java:986) ~[ItemJoin-6.0.3-b905.jar:?]
        at me.RockinChaos.itemjoin.item.ItemDesigner.registerItems(ItemDesigner.java:102) ~[ItemJoin-6.0.3-b905.jar:?]
        at org.bukkit.craftbukkit.v1_20_R2.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.20.2.jar:git-Paper-201]
        at org.bukkit.craftbukkit.v1_20_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:480) ~[paper-1.20.2.jar:git-Paper-201]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1097) ~[paper-1.20.2.jar:git-Paper-201]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:315) ~[paper-1.20.2.jar:git-Paper-201]
        at java.lang.Thread.run(Thread.java:1583) ~[?:?]
```

**Comment by RockinChaos** [comment]
After looking at the API for a while it seems something major has changed with pulling skull textures via the Minecraft site.
Working on a fix for this now.

**Comment by RockinChaos** [comment]
Okay, so it seems there were **a lot** of changes to Mojang's GameProfile. Took quite a bit of tinkering to fix, the latest snapshot (experimental) should resolve the problem.

Link; https://ci.craftationgaming.com/job/ItemJoin/906/
**Remember to remove your old ItemJoin.jar file**

**Comment by SlimeDog** [comment]
Confirmed fixed with build 906 on Paper 1.20.2-207. Thanks.
